VmAgent.Core/Interfaces/ProcessRunner.cs

```diff
@@ -66,16 +72,46 @@ public override Task<SessionHostInfo> CreateAndStart(int instanceNumber, GameRes
                 _logger.LogException($"Failed to start process based host with instance number {instanceNumber}", exception);
                 sessionHostManager.RemoveSessionHost(sessionHostUniqueId);
                 sessionHost = null;
+                string exceptionMessage = exception.ToString();
+                CollectLogs(logFolderPathOnVm, exceptionMessage);
             }
 
             return Task.FromResult(sessionHost);
         }
 
-        public override Task CollectLogs(string processId, string logsFolder, ISessionHostManager sessionHostManager)
+        public override Task CollectLogs(string id, string logsFolder, ISessionHostManager sessionHostManager)
+        {
+            return Task.CompletedTask;
+        }
+
+        public override Task CollectLogs(string logsFolder, string exceptionMessage)
```

Comment:
Could this method be called `CreateExceptionLog` or something? It's not really doing the same thing as the existing `CollectLogs` method, and if i'm not mistaken the existing method will still get called after this.